back/src/companies/shield-tree.ts

```diff
@@ -0,0 +1,15 @@
+import { chain } from "graphql-shield";
+import { isAuthenticated, isCompanyAdmin } from "../common/rules";
+import { createCompanySchema, createUploadLinkSchema } from "./rules/schema";
+
+export default {
+  Query: {
+    favorites: isAuthenticated
+  },
+  Mutation: {
+    updateCompany: isCompanyAdmin,
+    renewSecurityCode: isCompanyAdmin,
+    createCompany: chain(createCompanySchema, isAuthenticated),
```

Comment:
"The left-most rule is executed first.". Est-ce qu'on ne devrait pas vérifier les permissions avant la validation des inputs ?

Reply:
Je ne pense pas. Dans certains cas, si on a un mauvais input on ne sera meme pas capable de calculer les permissions (id manquant sur lequel on ferait des checks de droit par exemple)